src/gtc/dace/oir_to_dace.py

```diff
@@ -87,15 +87,20 @@ def make_output_dace_subset(self, node, field):
 
             return self._make_dace_subset(local_access_info, field)
 
-        def _make_dace_subset(self, local_access_info, field):
+        def _make_dace_subset(self, local_access_info, field) -> dace.subsets.Range:
             global_access_info = self.access_infos[field]
             return make_dace_subset(
                 global_access_info, local_access_info, self.decls[field].data_dims
             )
 
+        @property
+        def last_state(self) -> Optional[dace.SDFGState]:
+            nodes = self.sdfg.states()
+            return nodes[-1] if nodes else None
```

Comment:
I'm not sure if we have guarantee here that `sdfg.states()[-1]` is always the one we want.

Reply:
This would work I think:
```python
        @property
        def last_state(self) -> Optional[dace.SDFGState]:
            res = [node for node in self.sdfg.nodes() if self.sdfg.out_degree(node) == 0]
            if len(res) == 1:
                return res[0]
            else:
                assert len(res)==0
                return None
```
That's subject to quadratic complexity in the number of states though.

Reply:
Ah right. It does look like SDFG.nodes() is an OrderedDict so it happens to work out in this case, but we probably shouldn't rely on it. I'll switch that back.